File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;

        while (r > l) {
            if (numbers[r] + numbers[l] > target) {
                r--;
            }
            else if (numbers[r] + numbers[l] < target) {
                l++;
            }
            else {
                vector<int> output = {l + 1, r + 1};
                return output;
            }
        }
        
    }
};

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-2.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        distances = []
        output = []
        

        for x, y in points:
            dist = (x)**2 + (y)**2
            distances.append([dist, x, y])

        heapq.heapify(distances)

        for _ in range(k):
            dist, x, y = heapq.heappop(distances)
            output.append([x, y])

        return output

        